compiler/rustc_trait_selection/src/solve/project_goals.rs

```diff
@@ -70,23 +68,22 @@ impl<'tcx> EvalCtxt<'_, 'tcx> {
     /// This is the case if the `term` is an inference variable in the innermost universe
     /// and does not occur in any other part of the predicate.
     fn term_is_fully_unconstrained(&self, goal: Goal<'tcx, ProjectionPredicate<'tcx>>) -> bool {
```

Comment:
idk how to best explain this, but this function feels like a utility function, not like solver behavior. as far as the solver is concerned, this function is just a black box.

With this, and the fact that the rest solver shouldn't use `probe_x_var`, I would prefer us to move this function in `eval_ctxt` and remove the `probe_x_var` functions from `EvalCtxt`